### PR TITLE
Add mapper generator

### PR DESCRIPTION
**Pull request by rvsia**
When user runs

```bash
yarn generate-template
```

It will ask him for a mapper name and it will automatically create the project folder.

generator

### CODE REVIEW

File: templates/component-mapper/demo/demo-schemas/miq-schema.js [hunk 1]
```diff
@@ -0,0 +1,1949 @@
+/* eslint-disable camelcase */
+const input = {
```
Comment (Hyperkid123) [inline templates/component-mapper/demo/demo-schemas/miq-schema.js:2]: I don't think it's necessary to copy this schema. I think we can even remove it completely

File: templates/component-mapper/demo/demo-schemas/widget-schema.js [hunk 2]
```diff
@@ -0,0 +1,504 @@
+/*eslint camelcase: "off" */
```
Comment (Hyperkid123) [inline templates/component-mapper/demo/demo-schemas/widget-schema.js:1]: We can also remove this file

File: templates/component-mapper/demo/demo-schemas/wizard-schema.js [hunk 3]
```diff
@@ -0,0 +1,448 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
+import { Button } from '@patternfly/react-core';
```
Comment (Hyperkid123) [inline templates/component-mapper/demo/demo-schemas/wizard-schema.js:4]: Patternfly

File: templates/component-mapper/package.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
+{
+  "name": "@data-driven-forms/{{componentmapper}}-component-mapper",
+  "version": "0.4.2",
```
Comment (Hyperkid123) [inline templates/component-mapper/package.json:3]: This version number should be probably set to 0

File: templates/component-mapper/package.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    "javascript"
+  ],
+  "devDependencies": {
+    "@babel/core": "^7.1.6",
+    "@babel/plugin-proposal-class-properties": "^7.1.0",
+    "@babel/plugin-syntax-dynamic-import": "^7.0.0",
+    "@babel/preset-env": "^7.1.6",
+    "@babel/preset-react": "^7.0.0",
+    "@data-driven-forms/react-form-renderer": "*",
+    "@patternfly/react-core": "^3.153.13",
```
Comment (Hyperkid123) [inline templates/component-mapper/package.json:32]: PF reference

File: templates/component-mapper/package.json [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    "webpack": "^4.25.1",
+    "webpack-cli": "^3.1.2",
+    "webpack-dev-server": "^3.1.10",
+    "webpack-merge": "^4.1.4"
+  },
+  "peerDependencies": {
+  },
+  "dependencies": {
+  }
+}
```
Comment (Hyperkid123) [inline templates/component-mapper/package.json:79]: new line

File: templates/component-mapper/src/src/files/checkbox.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const MultipleChoiceList = () => 'MultipleChoiceList';
```
Comment (Hyperkid123) [inline templates/component-mapper/src/src/files/checkbox.js:4]: I would definitely prepare example using the common multiple-choice component from common package

File: templates/component-mapper/src/src/files/date-picker.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const DatePicker = () => 'MultipleChoiceList';
```
Comment (Hyperkid123) [inline templates/component-mapper/src/src/files/date-picker.js:4]: Wrong string. I think we can also think about using simple JSX markup using input types to show of these trivial components.

File: templates/component-mapper/src/src/files/field-array.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const FieldArray = () => 'FieldArray';
```
Comment (Hyperkid123) [inline templates/component-mapper/src/src/files/field-array.js:4]: Same here I would show of the common package.

File: templates/component-mapper/src/src/files/form-template.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
+};
+
+export const Title = ({ children }) => <h1>{children}</h1>;
+
+Title.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
+};
+
+export const Description = ({ children }) => <span>{children}</span>;
```
Comment (Hyperkid123) [inline templates/component-mapper/src/src/files/form-template.js:24]: `<p></p>` span is inline

File: templates/component-mapper/src/src/files/form-template.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+Description.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
+};
+
+export const Form = ({ children, ...props }) => <form {...props}>{children}</form>;
```
Comment (Hyperkid123) [inline templates/component-mapper/src/src/files/form-template.js:30]: `noValidate` is missing. The form will prioritize HTML validation over ours.

File: templates/component-mapper/src/src/files/radio.js [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Radio = () => 'Radio';
```
Comment (Hyperkid123) [inline templates/component-mapper/src/src/files/radio.js:4]: Again we can use some basic implementation to showcase what props and how to use them in order to get this working. Mainly the onChange is important

File: templates/component-mapper/src/src/files/select.js [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Select = () => 'Select';
```
Comment (Hyperkid123) [inline templates/component-mapper/src/src/files/select.js:4]: ^^

File: templates/component-mapper/package.json (hunk 6 above)
Comment (rvsia) [reply templates/component-mapper/package.json:79]: it's a json file :D

**Comment by rvsia** [comment]
@Hyperkid123 changed, it creates a fully functional mapper now. All components should work.

File: templates/component-mapper/src/files/checkbox.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+  children: PropTypes.node
+};
+
+const SingleCheckbox = (props) => {
+  const { input, isDisabled, label, name } = useFieldApi({ ...props, type: 'checkbox' });
+
+  return (
+    <React.Fragment>
+      <label htmlFor={name}>{label}</label>
+      <input {...input} type="checkbox" disabled={isDisabled}/>
```
Comment (Hyperkid123) [inline templates/component-mapper/src/files/checkbox.js:24]: you are missing `id={name}` for the label to work

**Pull request merged**